### PR TITLE
Add solana-stake-accounts CLI tool

### PR DESCRIPTION
**Pull request by garious**
#### Problem

Managing multiple derived stake accounts is tedious.

#### Summary of Changes

Add a new CLI tool, `solana-stake-accounts` for operating on all derived accounts with single invocations.

### CODE REVIEW

**Comment by garious** [comment]
@CriesofCarrots, @danpaul000, I'm thinking of stuffing this into v1.1.0 as a "preview release". Is there anything here particularly objectionable or can we do a post-merge review?

File: docs/src/cli/manage-stake-accounts.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Manage Stake Accounts
+
+If you want to delegate stake to many different validators, you will need
+to create a separate stake account for each. If you follow the convention
+of creating the first stake account at seed "0", the second at "1", the
+third at "2", and so on, then the `solana-stake-accounts` tool will allow
+you to operate on all accounts with single innovations. You can use it to
```
Comment (CriesofCarrots) [inline docs/src/cli/manage-stake-accounts.md]: This is a strange usage of "innovations". Do you mean "invocations"?

**Comment by CriesofCarrots** [comment]
> @CriesofCarrots, @danpaul000, I'm thinking of stuffing this into v1.1.0 as a "preview release". Is there anything here particularly objectionable or can we do a post-merge review?

I've only read the markdown so far, but I'm okay with that.

File: docs/src/cli/manage-stake-accounts.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# Manage Stake Accounts
+
+If you want to delegate stake to many different validators, you will need
+to create a separate stake account for each. If you follow the convention
+of creating the first stake account at seed "0", the second at "1", the
+third at "2", and so on, then the `solana-stake-accounts` tool will allow
```
Comment (danpaul000) [inline docs/src/cli/manage-stake-accounts.md]: If this tool can be used to create derived accounts, do we need to expose the "0", "1" seed string logic?

File: docs/src/cli/manage-stake-accounts.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+new authorities.
+
+## Usage
+
+### Create a stake account
+
+Create and fund a derived stake account at the stake authority public key:
+
+```bash
+solana-stake-accounts new <SENDER_KEYPAIR> <BASE_KEYPAIR> <AMOUNT> \
```
Comment (danpaul000) [inline docs/src/cli/manage-stake-accounts.md]: Does SENDER == FUNDING?  FUNDING_KEYPAIR makes more sense to me.  And BASE_KEYPAIR is the derivation root?

File: docs/src/cli/manage-stake-accounts.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+Sum the balance of dervied stake accounts:
+
+```bash
+solana-stake-accounts balance <BASE_PUBKEY> --num-accounts <NUMBER>
+```
+
+### Get stake account public keys
+
+List the public key of each stake account derived from the given public key:
```
Comment (danpaul000) [inline docs/src/cli/manage-stake-accounts.md]: s/public key/address/

File: docs/src/cli/manage-stake-accounts.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+```bash
+solana-stake-accounts pubkeys <BASE_PUBKEY> --num-accounts <NUMBER>
```
Comment (danpaul000) [inline docs/src/cli/manage-stake-accounts.md]: s/pubkeys/addresses/

File: docs/src/cli/manage-stake-accounts.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+```
+
+### Set new authorities
+
+Set new authorities on each derived stake account:
+
+```bash
+solana-stake-accounts authorize <BASE_PUBKEY> \
+    --stake-authority <KEYPAIR> --withdraw-authority <KEYPAIR> \
+    --new-stake-authority <KEYPAIR> --new-withdraw-authority <PUBKEY> \
```
Comment (danpaul000) [inline docs/src/cli/manage-stake-accounts.md]: I think this should be <PUBKEY>

File: docs/src/cli/manage-stake-accounts.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    --stake-authority <KEYPAIR> --num-accounts <NUMBER>
+```
+
+To atomically rebase and authorize each stake account, use the 'move'
+command:
+
+```bash
+solana-stake-accounts move <BASE_PUBKEY> <NEW_BASE_KEYPAIR> \
+    --stake-authority <KEYPAIR> --withdraw-authority <KEYPAIR> \
+    --new-stake-authority <KEYPAIR> --new-withdraw-authority <PUBKEY> \
```
Comment (danpaul000) [inline docs/src/cli/manage-stake-accounts.md]: PUBKEY not KEYPAIR for --new-stake-authority?

File: docs/src/cli/manage-stake-accounts.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+```
+
+### Relocate stake accounts
+
+Relocate stake accounts:
+
+```bash
+solana-stake-accounts rebase <BASE_PUBKEY> <NEW_BASE_KEYPAIR> \
+    --stake-authority <KEYPAIR> --num-accounts <NUMBER>
+```
```
Comment (danpaul000) [inline docs/src/cli/manage-stake-accounts.md]: This only works for non-delegated stake, right?  We'd need the --withdraw-authority as well, also I believe?

File: docs/src/cli/manage-stake-accounts.md (hunk 2 above)
Comment (garious) [reply docs/src/cli/manage-stake-accounts.md]: Lots of features can be added.  In this case, it'd be an `--index` option to the `new` command.  All the other commands operate on *all* derived accounts.  The tool isn't meant to redirect existing commands to a derived account.

**Comment by danpaul000** [comment]
> @CriesofCarrots, @danpaul000, I'm thinking of stuffing this into v1.1.0 as a "preview release". Is there anything here particularly objectionable or can we do a post-merge review?

Okay, that's fine.

File: docs/src/cli/manage-stake-accounts.md (hunk 3 above)
Comment (garious) [reply docs/src/cli/manage-stake-accounts.md]: "sender" is the counterpart to a "recipient" in `solana transfer`.   I'll think about it.  Mostly about where we want words with an "er" suffix or "ing".

File: docs/src/cli/manage-stake-accounts.md (hunk 7 above)
Comment (garious) [reply docs/src/cli/manage-stake-accounts.md]: wow, good eye!

File: docs/src/cli/manage-stake-accounts.md (hunk 8 above)
Comment (garious) [reply docs/src/cli/manage-stake-accounts.md]: It uses `split-stake`, so only needs the `--stake-authority`.

File: docs/src/cli/manage-stake-accounts.md (hunk 1 above)
Comment (garious) [reply docs/src/cli/manage-stake-accounts.md]: Yes, fixed

File: docs/src/cli/manage-stake-accounts.md (hunk 3 above)
Comment (garious) [reply docs/src/cli/manage-stake-accounts.md]: Went with FUNDING

File: docs/src/cli/manage-stake-accounts.md (hunk 4 above)
Comment (garious) [reply docs/src/cli/manage-stake-accounts.md]: Fixed

File: docs/src/cli/manage-stake-accounts.md (hunk 5 above)
Comment (garious) [reply docs/src/cli/manage-stake-accounts.md]: Fixed

File: docs/src/cli/manage-stake-accounts.md (hunk 6 above)
Comment (garious) [reply docs/src/cli/manage-stake-accounts.md]: should be `<PUBKEY>`, yes. nice catch

File: docs/src/cli/manage-stake-accounts.md (hunk 7 above)
Comment (garious) [reply docs/src/cli/manage-stake-accounts.md]: Fixed

File: docs/src/cli/manage-stake-accounts.md (hunk 2 above)
Comment (garious) [reply docs/src/cli/manage-stake-accounts.md]: Okay, added `solana-stake-accounts new --index <NUMBER>`

**Pull request merged**